python-avd/pyavd/_eos_designs/structured_config/overlay/router_bgp.py

```diff
@@ -24,457 +26,365 @@ class RouterBgpMixin(Protocol):
+            self.structured_config.router_bgp.address_family_evpn.neighbors.append_new(
+                ip_address=self._wan_ha_peer_vtep_ip(),
+                activate=True,
+                encapsulation=self.inputs.wan_encapsulation,
+            )
+
+    def _set_address_family_ipv4_sr_te(self: AvdStructuredConfigOverlayProtocol) -> None:
+        """Set the structured config for IPv4 SR-TE address family."""
         if not self.shared_utils.is_cv_pathfinder_router:
-            return None
+            return
 
-        address_family_ipv4_sr_te = {
-            "peer_groups": [
-                {
-                    "name": self.inputs.bgp_peer_groups.wan_overlay_peers.name,
-                    "activate": True,
-                },
-            ],
-        }
+        peer_groups = self.structured_config.router_bgp.address_family_ipv4_sr_te.peer_groups
 
-        if self._is_wan_server_with_peers:
-            address_family_ipv4_sr_te["peer_groups"].append({"name": self.inputs.bgp_peer_groups.wan_rr_overlay_peers.name, "activate": True})
+        peer_groups.append_new(name=self.inputs.bgp_peer_groups.wan_overlay_peers.name, activate=True)
 
-        return address_family_ipv4_sr_te
+        if self._is_wan_server_with_peers:
+            peer_groups.append_new(name=self.inputs.bgp_peer_groups.wan_rr_overlay_peers.name, activate=True)
 
-    def _address_family_link_state(self: AvdStructuredConfigOverlayProtocol) -> dict | None:
-        """Generate structured config for link-state address family."""
+    def _set_address_family_link_state(self: AvdStructuredConfigOverlayProtocol) -> None:
+        """Set the structured config for link-state address family."""
         if not self.shared_utils.is_cv_pathfinder_router:
-            return None
+            return
 
-        address_family_link_state = {
-            "peer_groups": [
-                {
-                    "name": self.inputs.bgp_peer_groups.wan_overlay_peers.name,
-                    "activate": True,
-                },
-            ],
-        }
+        peer_groups = self.structured_config.router_bgp.address_family_link_state.peer_groups
+        peer_groups.append_new(name=self.inputs.bgp_peer_groups.wan_overlay_peers.name, activate=True)
 
         if self.shared_utils.is_cv_pathfinder_server:
-            address_family_link_state["path_selection"] = {"roles": {"consumer": True, "propagator": True}}
-            address_family_link_state["peer_groups"][0].update(
-                {
-                    "missing_policy": {
-                        "direction_out_action": "deny",
-                    },
-                },
-            )
+            self.structured_config.router_bgp.address_family_link_state.path_selection.roles._update(consumer=True, propagator=True)
+            peer_groups.obtain(self.inputs.bgp_peer_groups.wan_overlay_peers.name).missing_policy.direction_out_action = "deny"
```

Comment:
Still an obtain here. Please just create the peer group object instead of append_new above and then update it here and append below.

Reply:
Fixed! 
Apologies for missing this one.